File: Chapter02/Java/Proxim/app/src/main/java/net/zenconsult/android/model/Location.java
package net.zenconsult.android.model;

public class Location {

    private String identifier;
    private double latitude;
    private double longitude;

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    public void setIdentifier(String identifier) {
        this.identifier = identifier;
    }

    public String getIdentifier() {
        return identifier;
    }

    @Override
    public String toString() {
        StringBuilder ret = new StringBuilder();
        ret.append(getIdentifier());
        ret.append(getLatitude());
        ret.append(getLongitude());
        return ret.toString();
    }

    public byte[] getBytes() {
        return toString().getBytes();
    }
}

File: Chapter02/Kotlin/Proxim/app/src/main/java/net/zenconsult/android/model/Location.kt
package net.zenconsult.android.model

import java.lang.StringBuilder

class Location(
    var identifier: String?,
    var latitude: Double = 0.0,
    var longitude: Double = 0.0
) {

    constructor() : this("", 0.0, 0.0)

    override fun toString(): String {
        val ret = StringBuilder()
        ret.append(identifier ?: "")
        ret.append(latitude)
        ret.append(longitude)
        return ret.toString()
    }

    public fun getBytes(): ByteArray {
        return toString().toByteArray()
    }
}